src/main/scala/ru/d4team/chat/db/DBMigrator.scala

```diff
@@ -0,0 +1,51 @@
+package ru.d4team.chat.db
+
+import org.flywaydb.core.api.MigrationState
+import org.flywaydb.core.api.configuration.FluentConfiguration
+import ru.d4team.chat.config.PostgresConfig
+import zio._
+
+
+object DBMigrator {
+
+  def migrate: ZIO[PostgresConfig, Throwable, Unit] =
+    for {
+      config <- ZIO.service[PostgresConfig]
+      _ <- ZIO.logInfo(s"Starting the migration for host: ${config.url}")
+      count <- migrationEffect(config)
+      _ <- ZIO.logInfo(s"Successful migrations: $count")
+    } yield ()
```

Comment:
We can use Layer described for flywayConfig via DI. So we can simplify migrate method and remove migrationEffect method taking into account the previous comments, like this: 

```
  def migrate: ZIO[FluentConfiguration, Throwable, Unit] =
    for {
      flywayConfig <- ZIO.service[FluentConfiguration]
      _ <- ZIO.logInfo(s"Starting the migration for host: ${flywayConfig.getUrl}")
      _ <- logValidationErrorsIfAny(flywayConfig)
      _ <- ZIO.logInfo("Migrations validation successful")
      count <- ZIO.succeed(flywayConfig.load().migrate().migrationsExecuted)
      _ <- checkStatesForSuccess(flywayConfig)
      _ <- ZIO.logInfo(s"Successful migrations: $count")
    } yield ()
```

Reply:
thank you, I refactored this fragment